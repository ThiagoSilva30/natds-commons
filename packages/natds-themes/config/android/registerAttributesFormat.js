import fs from 'fs';
import path from 'path';
import Handlebars from 'handlebars';

const registerHtmlFormat = () => {
  const html = fs.readFileSync(path.resolve(__dirname, './templates/attributes.hbs')).toString();
  const htmlTemplate = Handlebars.compile(html);

  const formatter = dictionary => htmlTemplate({
    color: dictionary.properties.color,
  });

  return {
    formatter,
    name: 'android/attributes',
  };
};

export default registerHtmlFormat;
