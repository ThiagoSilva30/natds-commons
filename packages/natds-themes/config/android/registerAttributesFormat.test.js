import registerAttributesFormat from './registerAttributesFormat';

const dictionary = {
  properties: {
    color: {
      primary: {
        attributes: {},
        name: 'colorPrimary',
        original: {
          value: '#F091C9',
        },
        path: [
          'color',
          'primary',
        ],
        value: '#F091C9',
      },
    },
  },
};

describe('registerAttributesFormat', () => {
  it('should return the register config', () => {
    const register = registerAttributesFormat();

    expect(typeof register.formatter).toBe('function');
    expect(register.name).toBe('android/attributes');
  });

  it('should return the correct formatter template', () => {
    const register = registerAttributesFormat();

    expect(register.formatter(dictionary)).toMatchSnapshot();
  });
});
