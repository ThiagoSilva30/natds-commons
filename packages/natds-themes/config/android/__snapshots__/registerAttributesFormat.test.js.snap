// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`registerAttributesFormat should return the correct formatter template 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?>
<resources>
  <attr name=\\"colorPrimary\\" format=\\"color\\" />
</resources>
"
`;
